r.niche.similarity: address #897

veroandreo opened the pull request:
This small PR solves the errors reported in #897. The printing in the csv file does not look great though... not a python expert here

```
Statistic,Layer 1,Layer 2,value
"['Niche overlap (D)', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.651129981460896]"
"['Niche overlap (I)', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.8902500062842025]"
"['Correlation coeff', 'cl_prediction_maxent', 'vl_prediction_maxent', 0.494894]"
```

veroandreo committed:
Partially address #897

This small PR solves the Attribute error part reported in #897

veroandreo committed:
address the type error in printing

This solves the second error. The text file does not look great though...

petrasovaa commented on src/raster/r.niche.similarity/r.niche.similarity.py:
```diff
@@ -88,8 +87,8 @@ def tmpname(prefix):
     """
-    tmpf = prefix + str(uuid.uuid4())
-    tmpf = string.replace(tmpf, "-", "_")
+    tmpf = 'prefix{}'.format(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
```
Perhaps this function could be used instead:

https://grass.osgeo.org/grass82/manuals/libpython/script.html#script.utils.append_uuid

petrasovaa commented on src/raster/r.niche.similarity/r.niche.similarity.py:
```diff
@@ -219,7 +218,7 @@ def main(options, flags):
+        IND = [["Statistic", "Layer 1", "Layer 2", "value"], Dind, Iind, Cind]
         import csv
 
         with open(OPF, "wb") as f:
```
This should fix the other error:
```suggestion
        with open(OPF, "w") as f:
```

veroandreo replied:
yes, I had it locally and forgot to change it in the actual PR that I did online, thanks!

veroandreo committed:
Update src/raster/r.niche.similarity/r.niche.similarity.py

Co-authored-by: Anna Petrasova <kratochanna@gmail.com>

ecodiv committed:
update r.niche.similarity bugfix

Changes to solve the errors reported in #897. The printing in the csv file is fixed as well (was caused by earlier solution, reverted to old code used to print to csv).

ecodiv commented:
@veroandreo I edited your pull request, fixing the errors reported in https://github.com/OSGeo/grass-addons/issues/897, using the same code I have used in other addons to create temporary file names. It also fixes the printing to csv issue (by reverting to old code). For me this works, can you please check?

ecodiv reviewed (commented):
See proposed changes (pushed to this branch)

veroandreo commented:
works as expected, thanks for checking and fixing; merging
